community/logging/src/main/java/org/neo4j/logging/FormattedLog.java

```diff
@@ -203,19 +207,30 @@ protected void writeLog( PrintWriter out, String message )
         @Override
         protected void writeLog( PrintWriter out, String message, Throwable throwable )
         {
+            requireNonNull( message, "message" );
+
             lineStart( out );
             out.write( message );
-            out.write( ' ' );
-            out.write( throwable.getMessage() );
-            out.println();
-            throwable.printStackTrace( out );
+            if ( throwable != null )
+            {
+                if ( throwable.getMessage() != null )
+                {
+                    out.write( ' ' );
+                    out.write( throwable.getMessage() );
+                }
+                out.println();
+                throwable.printStackTrace( out );
+            }
         }
 
         @Override
         protected void writeLog( PrintWriter out, String format, Object[] arguments )
         {
+            requireNonNull( format, "format" );
+
+            String message = String.format( format, arguments );
```

Comment:
To make sure that message is valid before we start writing to `out`. If format is invalid `out.format` will throw `IllegalFormatException` and we'll have partially written message. This could probably be handled in `AbstractPrintWriterLogger` and it'll call simple `writeLog( PrintWriter out, String message )` if message is valid. wdyt?